src/Nethermind/Nethermind.Consensus/Tracing/GethStyleTracer.cs

```diff
@@ -126,9 +126,10 @@ public GethLikeTxTrace Trace(Hash256 blockHash, int txIndex, GethTraceOptions op
             _processor.Process(block, ProcessingOptions.Trace, blockTracer.WithCancellation(cancellationToken));
             return blockTracer.BuildResult().SingleOrDefault();
         }
-        finally
+        catch
```

Comment:
Shouldn't we always dispose the tracer, not only when there is an exception?

Reply:
So we actually need to dispose the js engines in the tracer and potentially arraypoollist. 
1. I've removed arryapoollist as js tracers are already so slow and so much overhead that it won't make any difference to have it here.
2. we don't want to dispose engine's here as the engine's can have result objects attached to them, we need to serialize those results objects and only then we will be able to dispose the engines. This should be done by our serialization logic.
3. Only on failure we want to make sure we disposed already created engines.